pandas/tests/indexing/multiindex/test_loc.py

```diff
@@ -659,3 +659,30 @@ def test_getitem_non_found_tuple():
     )
     with pytest.raises(KeyError, match=r"\(2\.0, 2\.0, 3\.0\)"):
         df.loc[(2.0, 2.0, 3.0)]
+
+
+def test_loc_setitem_indexer_differently_ordered():
+    # GH#34603
+    mi = MultiIndex.from_product([["a", "b"], [0, 1]])
+    df = DataFrame([[1, 2], [3, 4], [5, 6], [7, 8]], index=mi)
+
+    indexer = ("a", [1, 0])
```

Comment:
should `("a", slice(None, None, -1))` behave the same as `("a", [1, 0])`?

Reply:
(and if so, does it work in this PR, and if so, can you test it)

Reply:
This is actually a completly different bug. In case of MultiIndex and slice, we will run into 

```
        def convert_indexer(start, stop, step, indexer=indexer, codes=level_codes):
            # given the inputs and the codes/indexer, compute an indexer set
            # if we have a provided indexer, then this need not consider
            # the entire labels set

            r = np.arange(start, stop, step)
```

which returns an empty array with negative step size. We have to change start and stop in these cases. I will look into this and try to fix this as a follow up, because not really related to this.

But more generall: I would say ``("a", slice(None, None, -1))`` should return the same as ``("a", [1, 0])``

Reply:
Actually fixing this only makes sense when #37707 is merged, because this affects slices too. Otherwise we would build erroneous tests

Reply:
OK.  If theres not already an issue for this separate bug, can you make one (assuming you havent already made a PR for it)

Reply:
Have a PR in the pipeline. Will add tests when this is merged.